src/sentry/dynamic_sampling/utils.py

```diff
@@ -43,3 +54,54 @@ def generate_uniform_rule(project: Project) -> UniformRule:
         },
         "id": UNIFORM_RULE_RESERVED_ID,
     }
+
+
+class DynamicSamplingFeatureMultiplexer:
+    """
+    This class is used to route DS behaviour according to the feature flags listed here
+    Essentially the logic is as follows:
+    - The `organizations:server-side-sampling` feature flag is the main flag enabled for dynamic sampling both in
+    sentry and in relay
+    - The  `organizations:server-side-sampling-ui` feature flag is the flag that enables the old dynamic sampling
+    behaviour which needs to be supported for backwards compatibility but is deprecated
+    - The `organizations:dynamic-sampling-basic` feature flag is the flag that enables the new adaptive sampling
+    """
+
+    def __init__(self, project, request):
+        # Feature flag that informs us that relay is handling DS rules
+        self.allow_dynamic_sampling = features.has(
+            "organizations:server-side-sampling", project.organization, actor=request.user
+        )
+        # Feature flag that informs us that the org is on the new AM2 plan and thereby have adaptive sampling enabled
+        # ToDo(ahmed): This needs to be renamed to `organizations:dynamic-sampling`
+        self.current_dynamic_sampling = features.has(
+            "organizations:dynamic-sampling-basic", project.organization, actor=request.user
+        )
+        # Flag responsible to inform us if the org was in the original LA/EA Dynamic Sampling
+        # ToDo(ahmed): This needs to be renamed to `organizations:dynamic-sampling-deprecated`
+        self.deprecated_dynamic_sampling = features.has(
+            "organizations:server-side-sampling-ui", project.organization, actor=request.user
+        )
+
+    def is_on_dynamic_sampling_deprecated(self):
+        return (
+            self.allow_dynamic_sampling
+            and self.deprecated_dynamic_sampling
+            and not self.current_dynamic_sampling
+        )
+
+    def is_on_dynamic_sampling(self):
+        return self.allow_dynamic_sampling and self.current_dynamic_sampling
+
+    def get_user_biases(self, user_set_biases):
```

Comment:
```suggestion
    def get_user_biases_or_default(self, user_set_biases):
```

I would honestly be more specific that this function works with a fallback data structure, otherwise it is a bit unclear from the outside why you need to pass user biases to get back user biases. In addition, this method can be `static` because it doesn't use anything that is bound to the instance.

Reply:
An alternative implementation would be to automatically instantiate `DynamicSamplingFeatureMultiplexer` with the list of user biases because it seems like you will always call the `get_user_biases` function with the same user biases.

Reply:
i'm not sure that `_or_default` is a good idea, in python we do a lot of OR_DEFAULT everywhere.